### PR TITLE
Add CI windows testing

### PR DESCRIPTION
**Pull request by relf**
Tentative PR to get windows testing for linfa. 

I've removed the usage of the container `rustmath:mkl-rust` to factorize actions and as I have some doubt [this Docker image](https://hub.docker.com/r/rustmath/mkl/tags) will be maintained in the future. Moreover, it might also prepare the field for no-blas PR landing.

### CODE REVIEW

**Comment by relf** [comment]
The PR fails on windows as expected. Waiting for `mnist 0.5` windows bug fix.

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -4,19 +4,20 @@ name: Run Tests
...
       matrix:
         toolchain:
           - 1.54.0
           - stable
-
+        platform: [{ os: "ubuntu-18.04", 
+            rust-target: "x86_64-unknown-linux-gnu"},
+          { os: "windows-2019", 
+            rust-target: "x86_64-pc-windows-msvc"},
+        ]
```
Comment (YuhanLiin) [inline .github/workflows/testing.yml:20]: Fix up formatting

**Comment by YuhanLiin** [comment]
The container is definitely unmaintained, and the tests actually run faster without it. We should also parametrize the build checking step by the OS.

File: .github/workflows/testing.yml (hunk 1 above)
Comment (relf) [reply .github/workflows/testing.yml]: Done

**Comment by relf** [comment]
I fiddled quite a bit with the github actions yml. I did not succeed in making `include` work with `os` definition, so in the end I removed it and handled `nightly` like `1.54` and `stable`.

**Comment by relf** [comment]
@YuhanLiin, @quietlychris, I've finally disabled the `mnist` example in `linfa-tsne` (it can be enabled again once `mnist` > 0.5 is released).

**Review by YuhanLiin — APPROVED** [review]

**Comment by YuhanLiin** [comment]
We can keep the issue open until the issue is fixed in `mnist`

**Comment by relf** [comment]
Ok, I've removed the text in the PR description which would triggered the closing of the related issue #209. 

Note: once this PR is merged, required status checks have to be adjusted.

**Pull request merged**